x-pack/plugins/serverless_search/public/application/components/index_documents/documents.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import {
+  DocumentList,
+  DocumentsOverview,
+  INDEX_DOCUMENTS_META_DEFAULT,
+} from '@kbn/search-index-documents';
+
+import { HttpSetup } from '@kbn/core-http-browser';
+import { useQuery } from '@tanstack/react-query';
+import { IndicesGetMappingIndexMappingRecord } from '@elastic/elasticsearch/lib/api/types';
+import { i18n } from '@kbn/i18n';
+import { useIndexDocumentSearch } from '../../hooks/api/use_index_documents';
+
+const DEFAULT_PAGINATION = {
+  pageIndex: INDEX_DOCUMENTS_META_DEFAULT.pageIndex,
+  pageSize: INDEX_DOCUMENTS_META_DEFAULT.pageSize,
+  totalItemCount: INDEX_DOCUMENTS_META_DEFAULT.totalItemCount,
+};
+
+interface IndexDocumentsProps {
+  indexName: string;
+  http: HttpSetup;
+}
+
+export const IndexDocuments: React.FC<IndexDocumentsProps> = ({ http, indexName }) => {
+  const [pagination, setPagination] = useState(DEFAULT_PAGINATION);
+  const [searchQuery, setSearchQuery] = useState('');
+  const searchQueryCallback = (query: string) => {
+    setSearchQuery(query);
+  };
+  const { results: indexDocuments, meta: documentsMeta } = useIndexDocumentSearch(
+    indexName,
+    pagination,
+    http,
+    searchQuery
+  );
+
+  const { data: mappingData } = useQuery({
```

Comment:
we should probably write an api hook for this query too, like you did for the `useIndexDocumentSearch`

Reply:
I initially had changes for this. But thought we might not be using this hook in any other components.

Reply:
Even if we're aren't using it elsewhere it's good to have the API hook so that there is mostly one place where API calls are defined.

There is probably api calls elsewhere that needs to be moved as well, but new ones we should try to always define in the api folder IMO.